metaseq/checkpoint_utils.py

```diff
@@ -163,23 +163,44 @@ def _launch_sbatch_for_checkpoint_copy(
             nfs_dir += "/"
         copy_script_dir = os.path.join(nfs_dir, cp_script_dir)
         os.makedirs(copy_script_dir, exist_ok=True)
-        sbatch_run_file = os.path.join(
-            copy_script_dir, f"_cp_sbatch_script_{num_update}.sh"
+
+        all_slurm_nodes = os.environ["SLURM_NODELIST"]
+        slurm_node_list = (
+            subprocess.check_output(
+                f"scontrol show hostnames {all_slurm_nodes}", shell=True
+            )
+            .decode()
+            .strip()
+            .split("\n")
         )
-        slurm_nodes = os.environ["SLURM_NODELIST"]
-        with open(sbatch_run_file, "w") as f:
-            f.write(
-                SBATCH_CHECKPOINT_COPY_CMD.format(
-                    oss_dir=oss_dir,
-                    slurm_nodes=slurm_nodes,
-                    local_dir=cfg.save_dir,
-                    num_files=num_files_per_host,
-                    nfs_dir=nfs_upload_path[4:],
-                    cp_script_dir=cp_script_dir,
-                    num_update=num_update,
+        num_hosts_per_chunk = 8
+        for idx in range(
+            0, len(slurm_node_list), num_hosts_per_chunk
+        ):  # each job ssh's to 8 hosts
+            node_list = ",".join(slurm_node_list[idx : idx + num_hosts_per_chunk])
+            host_list = (
+                subprocess.check_output(
+                    f"scontrol show hostlist {node_list}", shell=True
                 )
+                .decode()
+                .strip()
             )
-        subprocess.call([f"sbatch {sbatch_run_file}"], shell=True)
+            sbatch_run_file = os.path.join(
+                copy_script_dir, f"_cp_sbatch_script_{num_update}_{host_list}.sh"
```

Comment:
Does this work? Maybe {idx} instead of {host_list} in the script name to be safe?

Reply:
Yup! Just have quotes around filenames at the end of this.